### PR TITLE
add scons 4.2.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **scons/4.2.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
> Thanks for the fast analysis! It's much appreciated! I changed the `readlink` to a loop. Let's see how it goes.

at least locally, it passes with your changes

**Comment by SSE4** [comment]
```
ImportError: cannot import name DEVNULL
```
seems like it happens because on ci `python` is 2.7...

**Comment by madebr** [comment]
Good find, so let's first try `python3`, and then fall back to `python`.

File: recipes/scons/all/conanfile.py [hunk 1]
```diff
@@ -21,73 +19,83 @@ class SConsConan(ConanFile):
...
+              done
+              local ourrealpath="$PWD/$(basename "$1")"
+              cd "$startpwd"
+              echo "$ourrealpath"
+            )
+
+            currentdir="$(dirname "$(realpath "$0")")"
+
+            export PYTHONPATH="$currentdir/../res:$PYTHONPATH"
+            exec ${PYTHON:-python} "$currentdir/../res/SCons/__main__.py" $*
```
Comment (SSE4) [inline recipes/scons/all/conanfile.py:67]:
```suggestion
            exec ${PYTHON:-python3} "$currentdir/../res/SCons/__main__.py" $*
```
let's try python3

File: recipes/scons/all/conanfile.py [hunk 2]
```diff
@@ -21,73 +19,83 @@ class SConsConan(ConanFile):
...
-            self.run("{} setup.py --requires".format(self._python_executable), output=output)
-            # Workaround for log.print_run_commands = True/False
-            # This requires log.run_to_output = True
-            if not (output.getvalue().strip().splitlines() or ["-"])[-1].startswith("-"):
-                raise ConanException("scons has a requirement")
-            self.run("{} setup.py build".format(self._python_executable))
+    def _scons_cmd(self):
+        return os.path.join(self.package_folder, "bin", "scons.cmd")
 
     def package(self):
```
Comment (madebr) [inline recipes/scons/all/conanfile.py:38]:
```suggestion
        return os.path.join(self.package_folder, "bin", "scons.cmd")

    def package_id(self):
        self.info.header_only()

    def package(self):
```

File: recipes/scons/all/conanfile.py [hunk 3]
```diff
@@ -21,73 +19,86 @@ class SConsConan(ConanFile):
...
+
         bindir = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment var: {}".format(bindir))
         self.env_info.PATH.append(bindir)
 
-        scons_pythonpath = os.path.join(self.package_folder, "lib", "site-packages", "scons")
-        self.output.info("Appending PYTHONPATH environment var: {}".format(scons_pythonpath))
-        self.env_info.PYTHONPATH.append(os.path.join(self.package_folder, "lib", "site-packages", "scons"))
+        if self.settings.os == "Windows":
+            scons_bin = os.path.join(bindir, "scons.bat")
```
Comment (madebr) [inline recipes/scons/all/conanfile.py:99]:
```suggestion
            scons_bin = os.path.join(bindir, "scons.cmd")
```

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Comment by ericLemanissier** [comment]
Thank you for the help! I didn't do anything here :D

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
team

**Pull request merged**